python/ray/tests/test_failure.py

```diff
@@ -799,3 +800,173 @@ def test(self):
+            object_manager_repeated_push_delay_ms=1000,
+            initial_reconstruction_timeout_milliseconds=200,
+        )
+    }, {
+        "object_store_memory": 150 * 1024 * 1024,
+        "num_nodes": 1,
+        "num_cpus": 10,
+    }],
+    indirect=True)
+def test_direct_call_serialized_id_eviction(ray_start_cluster):
+    object_store_memory = 150 * 1024 * 1024
+    num_objects = 10
+
+    @ray.remote
+    def create_object(size):
+        print("create")
+        # Sleep a bit to allow getters to time out.
+        time.sleep(1)
+        print("create done")
+        return np.random.rand(size)
+
+    @ray.remote
+    def get(obj_ids):
+        print("get", obj_ids)
+        obj_id = obj_ids[0]
+        val = ray.get(obj_id)
+        print("get done", obj_ids)
+        assert isinstance(val, np.ndarray), val
+
+    # Submit enough tasks so that they exceed the size of the object store.
+    objects = []
+    for _ in range(num_objects):
+        obj = create_object.remote(object_store_memory // num_objects)
+        objects.append(obj)
+
+    tasks = [get.remote([obj]) for obj in objects]
+
+    # Get each object again. At this point, the earlier objects should have
+    # been evicted.
+    num_evicted, num_success = 0, 0
+    for task in tasks:
+        try:
+            val = ray.get(task)
+            assert val is None
+            num_success += 1
+        except ray.exceptions.UnreconstructableError:
+            num_evicted += 1
+        except ray.exceptions.RayTaskError:
+            num_evicted += 1
+    # Some objects should have been evicted.
+    assert num_evicted > 0
+
+
+@pytest.mark.skip(
+    "Uncomment once eviction errors for serialized IDs are implemented")
+@pytest.mark.parametrize(
+    "ray_start_cluster", [{
+        "num_nodes": 2,
+        "num_cpus": 10,
+    }, {
+        "num_nodes": 1,
+        "num_cpus": 20,
+    }],
+    indirect=True)
+def test_direct_call_serialized_id(ray_start_cluster):
+    @ray.remote
+    def create_object():
+        print("create")
+        # Sleep a bit to allow getters to time out.
+        time.sleep(1)
+        print("create done")
+        return np.random.rand(1)
+
+    @ray.remote
+    def get(obj_ids):
+        print("get", obj_ids)
+        obj_id = obj_ids[0]
+        val = ray.get(obj_id)
+        print("get done", obj_ids)
+        assert isinstance(val, np.ndarray), val
+
+    # Submit enough tasks so that they exceed the size of the object store.
+    objects = []
+    num_objects = 10
+    for _ in range(num_objects):
+        obj = create_object.remote()
+        objects.append(obj)
+
+    tasks = [get.remote([obj]) for obj in objects]
+    for obj in objects:
+        try:
+            ray.get(obj)
+        except ray.exceptions.UnreconstructableError:
+            pass
+        except ray.exceptions.RayTaskError:
+            pass
+
+    # Get each object again. At this point, the earlier objects should have
+    # been evicted.
```

Comment:
Would it be easier to use ray.internal.delete() so the tests can instead use a single object without needing to fill the store?

Reply:
Good idea, I'll try it.